### PR TITLE
allow to specify a proxy server for package downloads

### PR DESCRIPTION
**Pull request by xavvo**
sometimes we need to specify a proxy to download the consul package. So we can just pass the corresponding puppet-archive parameter.

### CODE REVIEW

**Comment by solarkennedy** [comment]
Is the use-case that only this *specific* download needs use a proxy server, but nothing else in the catalog?

Can you read https://github.com/solarkennedy/puppet-consul/issues/269 first and see if it is relevant?

**Comment by xavvo** [comment]
Thank you for the issue pointer. I also think it is cleaner to specify the proxy to the consul module instead of the archive class, because this would create a hidden dependency that breaks if you happen to use something different to download the consul package.

And yes, we do use the archive module for other use cases that do not require the proxy.

File: manifests/install.pp [hunk 1]
```diff
@@ -37,6 +37,7 @@
       -> archive { "${install_path}/consul-${consul::version}.${consul::download_extension}":
         ensure       => present,
         source       => $::consul::real_download_url,
+        proxy_server => $consul::proxy_server,
```
Comment (solarkennedy) [inline manifests/install.pp]: can you add the leading `$::` for consistency ?

**Comment by solarkennedy** [comment]
Resolves #269, cc @walkamongus

**Comment by walkamongus** [comment]
@solarkennedy thanks for the followup!

**Pull request merged**